data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwardingProcessorDecorator.java

```diff
@@ -67,7 +67,16 @@ public static List<Processor> decorateProcessors(
                     "Peer Forwarder Plugin: %s cannot have empty identification keys." + pluginId);
         }
 
-        final PeerForwarder peerForwarder = peerForwarderProvider.register(pipelineName, pluginId, identificationKeys, pipelineWorkerThreads);
+        long localModeProcessors = processors.stream()
+            .filter(processor -> ((RequiresPeerForwarding)processor).isForLocalProcessingOnly(null))
+            .count();
+        final PeerForwarder peerForwarder;
+        if (localModeProcessors < processors.size()) {
```

Comment:
I think one scenario we'd want to support is to have a single non-local aggregation processor and then multiple local. The reason is that you could use the same identification keys. The first aggregate could be non-local and that would get all the events onto the right node. Then you can run locally.

This logic does not support that.

Reply:
>>>single non-local aggregation processor 

We should move away from peer forwarders now. Like use Centralized S3 buffering mechanism for non-local aggregations.